File: lab/closestPair.py
import sys


def closest_pair():
    """Input the string of array"""
    s = str(input("Input the string of array: "))
    s_list = s.split(",")
    num_list = con_string(s_list)
    find_closest(num_list)


def find_closest(num_list):
    """Find the closest pair"""
    closest = sys.maxsize
    subtract = [0]*len(num_list)
    for i in range(len(num_list)):
        if i < len(num_list)-1:
            temp = num_list[i]-num_list[i+1]
            if temp < 0:
                temp = -temp
            if temp < closest:
                 closest = temp
            subtract[i] = temp
    print_index(subtract, closest)


def print_index(subtract, closest):
    for i in range(len(subtract)):
        if subtract[i] is closest:
            print("Pair found at index ", i, " and ", (i+1))


def con_string(s_list):
    """Convert the string to the list of integer"""
    num_list = []
    for i in s_list:
        num_list.append(int(i))
    return num_list


closest_pair()

File: CS210FZ/lab/closestPair.java
import java.util.Scanner;

public class closestPair {
    // Find the closest pair
    public static void main(String[] args) {
        // Given an input string of a sequence
        // of integers from the console.
        Scanner in = new Scanner(System.in);

        // Create a String
        String s = in.nextLine();

        // Change the string to the integer array
        int[] num = createArray(s);

        // Find the closest pair of the adjacent element
        findClosest(num);
        in.close();
    }

    // Create an array of integer，
    // read the string and parse
    // it as an unsorted array of integers
    public static int[] createArray(String s){
        // Read the string and parse it
        // as an unsorted array of integers
        String[] stringNum = s.split(",");
        int len = stringNum.length;
        int[] num = new int[len];
        for (int i = 0; i < len; i++){
            num[i] = Integer.parseInt(stringNum[i]);
        }
        return num;
    }

    // Find the closest pair of the adjacent element
    public static void findClosest(int[] num){
        int len = num.length-1
                , min = Integer.MAX_VALUE
                , temp;
        int[] subtract = new int[len];
        for (int i = 0; i < len; i++){
            temp = num[i] - num[i+1];
           if (temp < 0) temp = -temp;
           if (temp < min) min = temp;
           subtract[i] = temp;
        }
        for (int i = 0; i < num.length-1; i++) {
            if(subtract[i] == min) System.out.println("Pair found at index "+ i +" and "+ (i+1));
        }
    }
}
